src/ci/github-actions/ci.yml

```diff
@@ -301,7 +301,7 @@ defaults:
 concurrency:
   # For a given workflow, if we push to the same PR, cancel all previous builds on that PR.
   # If the push is not attached to a PR, we will cancel all builds related to the same commit SHA.
-  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
```

Comment:
I think this is also not the behavior we want, because we *do* want to cancel concurrent auto builds, while we don't want to cancel concurrent try builds. I think we can use [this to check which branch we're in](https://docs.github.com/en/actions/learn-github-actions/expressions#example).

Reply:
Oh, I thought that auto builds happen in the same way as try builds. But yeah, it makes sense to "serialize" the auto branch. I guess that bors makes sure that it doesn't run in parallel at the moment?

Reply:
I have done a few more experiments and I suggest this expression as the concurrency key:
```yaml
${{ (github.ref == 'refs/heads/try' && github.sha) || github.ref }}
```
By default, we only allow one CI per branch. This will work both for auto and for PRs, because for PRs the ref will look like this: `refs/pull/<pr_number>/merge`. On top of that, we add an explicit exception for the `try` branch, which will use the commit SHA as the concurrency key, and thus it will allow concurrent builds.

Reply:
Sounds good.